nextcord/application_command.py

```diff
@@ -0,0 +1,883 @@
+                uncalled_args.pop(arg_data["name"])
+                kwargs[self.arguments[arg_data["name"]].functional_name] = \
+                    self.arguments[arg_data["name"]].handle_slash_argument(state, arg_data["value"], interaction)
+            else:
+                # TODO: Handle this better.
+                raise NotImplementedError(f"An argument was provided that wasn't already in the function, did you"
+                                          f"recently change it?\nRegistered Args: {self.arguments}, Discord-sent"
+                                          f"args: {interaction.data['options']}, broke on {arg_data}")
+        for uncalled_arg in uncalled_args.values():
+            kwargs[uncalled_arg.functional_name] = uncalled_arg.default
+        await self.invoke_slash(interaction, **kwargs)
+
+    async def invoke_slash(self, interaction: Interaction, **kwargs):
+        """|coro|
+
+        Invokes the callback with the kwargs given."""
+        if self.cog_parent:
+            await self.callback(self.cog_parent, interaction, **kwargs)
+        else:
+            await self.callback(interaction, **kwargs)
+
+    def error(self, coro):
+        # TODO: Parity with legacy commands.
+        raise NotImplementedError
+
+    @property
+    def payload(self) -> dict:
+        """:class:`dict`: Outputs a Discord "Application Command Option Structure" payload."""
+        self._analyze_content()
+        ret = {"type": self.type.value, "name": self.name, "description": self.description}
+        if self.choices:
+            ret["choices"] = [{key: value} for key, value in self.choices.items()]
+        if self.children:
+            ret["options"] = [child.payload for child in self.children.values()]
+        elif self.arguments:
+            ret["options"] = [argument.payload for argument in self.arguments.values()]
+        return ret
+
+    def subcommand(self, **kwargs):
+        """Creates a new subcommand off of this one."""
+        def decorator(func: Callable):
+            result = ApplicationSubcommand(func, self, ApplicationCommandOptionType.sub_command, **kwargs)
+            self.children[result.name] = result
+            return result
+        return decorator
+
+
+class ApplicationCommand(ApplicationSubcommand):
+    def __init__(self, callback: Callable, cmd_type: ApplicationCommandType,
+                 name: str = "", description: str = "", guild_ids: List[int] = None, force_global: bool = False,
+                 default_permission: Optional[bool] = None):
+        super().__init__(callback=callback, parent=None, cmd_type=cmd_type, name=name, description=description)
+        # Basic input checking.
+        if guild_ids is None:
+            guild_ids = []
+        self._state: Optional[ConnectionState] = None  # TODO: I thought there was a way around doing this, but *sigh*.
+        self._is_global: Optional[bool] = True if (guild_ids and force_global) or (not guild_ids) else False
```

Comment:
I *feel* like this could be ``... if force_global or not guild_ids ...`` but I'd like a second opinion / someone to actually test this for me.

Reasoning being that variable is used in both states within the if